pgtrigger/core.py

```diff
@@ -2,20 +2,57 @@
 import copy
 import hashlib
 import logging
+import threading
 
 from django.db import connection
 from django.db import models
 from django.db.models.expressions import Col
 from django.db.models.fields.related import RelatedField
 from django.db.models.sql import Query
 from django.db.models.sql.datastructures import BaseTable
+import pgconnection
 
 
 LOGGER = logging.getLogger('pgtrigger')
 
 
 # All registered triggers for each model
-registry = set()
+registry = {}
+
+# All triggers currently being ignored
+_ignore = threading.local()
+
+
+def _is_concurrent_statement(sql):
+    """
+    True if the sql statement is concurrent and cannot be ran in a transaction
+    """
+    sql = sql.strip().lower() if sql else ''
+    return sql.startswith('create') and 'concurrently' in sql
+
+
+def _inject_pgtrigger_ignore(sql, sql_vars, cursor):  # pragma: no cover
```

Comment:
This is basically copied directly from django-pghistory and modified to inject a variable for django-pgtrigger. I ignored testing of some of the named cursor execution stuff since it was already tested there. I can copy those tests over if you'd like